packages/token-bridge-sdk/src/hooks/useArbTokenBridge.ts

```diff
@@ -890,157 +883,86 @@ export const useArbTokenBridge = (
       throw new TokenDisabledError('Token currently disabled')
     }
 
-    bridgeTokensToAdd[l1Address.toLowerCase()] = {
+    const l1AddressLowerCased = l1Address.toLowerCase()
+    bridgeTokensToAdd[l1AddressLowerCased] = {
       name,
       type: TokenType.ERC20,
       symbol,
-      address: l1Address.toLowerCase(),
-      l2Address,
+      address: l1AddressLowerCased,
+      l2Address: l2Address?.toLowerCase(),
       decimals
     }
 
     setBridgeTokens(oldBridgeTokens => {
       return { ...oldBridgeTokens, ...bridgeTokensToAdd }
     })
-    setErc20Balances(oldBridgeBalances => {
-      const newBal = {
-        [l1Address.toLowerCase()]: {
-          balance: l1TokenBalance,
-          arbChainBalance: l2TokenBalance
-        }
-      }
-      return { ...oldBridgeBalances, ...newBal }
-    })
-    return l1Address
-  }
-
-  const expireCache = (): void => {
-    clearERC20Cache()
-    clearERC721Cache()
-  }
 
-  useEffect(() => {
-    const tokensToAdd = [
-      ...new Set([...defaultTokenList].map(t => t.toLocaleLowerCase()))
-    ].filter(tokenAddress => !tokenBlackList.includes(tokenAddress))
-    if (autoLoadCache) {
-      Promise.all(
-        tokensToAdd.map(address => {
-          return addToken(address).catch(err => {
-            console.warn(`invalid cache entry erc20 ${address}`)
-            console.warn(err)
-          })
-        })
-      ).then(values => {
-        setERC20Cache(values.filter((val): val is string => !!val))
+    updateErcL1Balance({
+      [l1Address]: l1TokenBalance
+    })
+    if (l2Address && l2TokenBalance) {
+      updateErcL2Balance({
+        [l2Address]: l2TokenBalance
       })
     }
-  }, [])
+    return l1Address
```

Comment:
I know this is already in production code but this async function `addToken` seems strange

it has only one usage `addToken(erc20L1Address)` in an async function but used without `await` before it, and its return value has never been utilised.... It looks like legacy code? We can probably not return anything?

might want to refactor it in another PR

Reply:
It's used in rest of the app as `token.add`, but result is never used, so it can be removed